docs/testing/integration-tests.md

```diff
@@ -0,0 +1,131 @@
+ensuring they don't break existing features.
+
+## Setting Up PostProposalCheck.sol
+
+The first step is to create a `PostProposalCheck.sol` contract, which serves as
+a base for your integration test contracts. This contract is responsable for
+deploying proposal contracts, executing them, and updating the address objects. We'll illustrate this with the Multisig example from our [Multisig Proposal Guide](../guides/multisig-proposal.md).
+
+```solidity
+pragma solidity 0.8.19;
+
+import "@forge-std/Test.sol";
+import { MULTISIG_01 } from "path/to/MULTISIG_01.sol";
+import { MULTISIG_02 } from "path/to/MULTISIG_02.sol";
+import { MULTISIG_03 } from "path/to/MULTISIG_03.sol";
+import { Addresses } from "@forge-proposal-simulator/addresses/Addresses.sol";
+import { Constants } from "@forge-proposal-simulator/utils/Constants.sol";
+import { TestSuite } from "@forge-proposal-simulator/test/TestSuite.sol";
+
+// @notice this is a helper contract to execute proposals before running integration tests.
+// @dev should be inherited by integration test contracts.
+contract MultisigPostProposalCheck is Test {
+    string public constant ADDRESSES_PATH = "./addresses/Addresses.json";
+    TestSuite public suite;
+    Addresses public addresses;
+
+    function setUp() public virtual {
+        // Create proposals contracts
+        MULTISIG_01 multisigProposal = new MULTISIG_01();
+        MULTISIG_02 multisigProposal2 = new MULTISIG_02();
+        MULTISIG_03 multisigProposal3 = new MULTISIG_03();
+
+        // Populate addresses array
+        address[] memory proposalsAddresses = new address[](3);
+        proposalsAddresses[0] = address(multisigProposal);
+        proposalsAddresses[1] = address(multisigProposal2);
+        proposalsAddresses[2] = address(multisigProposal3);
+
+        // Deploy TestSuite contract
+        suite = new TestSuite(ADDRESSES_PATH, proposalsAddresses);
+
+        // Set addresses object
+        addresses = suite.addresses();
+
+        // Verify if the multisig address is a contract; if is not (e.g. running on a empty blockchain node), etch Gnosis Safe bytecode onto it.
+        address multisig = addresses.getAddress("DEV_MULTISIG");
+        uint256 multisigSize;
+        assembly {
+            multisigSize := extcodesize(multisig)
+        }
+        if (multisigSize == 0) {
+            vm.etch(multisig, Constants.SAFE_BYTECODE);
+        }
+
+        suite.setDebug(true);
+        // Execute proposals
+        suite.testProposals();
+
+        // Proposals execution may change addresses, so we need to update the addresses object.
+        addresses = suite.addresses();
+    }
+}
+```
+
+## Creating Integration Test Contracts
+
+Next, we create the `MultisigProposalTest` contract, inheriting
+`MultisigPostProposalCheck`. Add your tests to this contract, and use the
+`addresses` object to access the addresses of the contracts deployed by the
+proposals.
+
+```solidity
```

Comment:
What do we think about moving these code examples into the `examples/` directory?